frank/utilities.py

```diff
@@ -231,58 +324,90 @@ def cut_data_by_baseline(u, v, vis, weights, cut_range):
 
     return u_cut, v_cut, vis_cut, weights_cut
 
-
-def apply_correction_to_weights(u, v, ReV, weights, nbins=300):
+def estimate_weights(u, v, V, nbins=300, log=True, use_median=False):
     r"""
-    Estimate and apply a correction factor to the data's weights by comparing
-    binnings of the real component of the visibilities under different
-    weightings. This is useful for mock datasets in which the weights are all
-    unity.
+    Estimate the weights using the variance of the binned visibilities.
+
+    The estimation is done assuming that the variation in each bin is dominated
+    by the noise. This will be true if:
+        1) The source is axi-symmetric,
+        2) The uv-points have been deprojected,
+        3) The bins are not too wide,
+    Otherwise the variance may be dominated by real variations in the
+    visibilities.
 
     Parameters
     ----------
     u, v : array, unit = :math:`\lambda`
-        u and v coordinates of observations
-
-    ReV : array, unit = Jy
-        Real component of observed visibilities
-
-    weights : array, unit = Jy^-2
-        Weights assigned to observed visibilities, of the form
-        :math:`1 / \sigma^2`
-
+        u and v coordinates of observations (deprojected).
+    V : array, unit = Jy
+        Obsersed visibility. If complex, the weights will be computed from the
+        average of the variance of the real and imaginary components, as in
+        CASA's statwt. Otherwise the variance of the real part is used.
     nbins : int, default=300
-        Number of bins used to construct the histograms
+        Number of bins used.
+    log : bool, default=True
+        If True, the uv bins will be constructed in log space, otherwise linear
+        spaced bins will be used.
+    use_median : bool, default=False
+        If True all of the weights will be set to the median of the variance
+        estimated across the bins. Otherwise, the baseline dependent variance
+        will be used.
 
     Returns
     -------
-    wcorr_estimate : float
-        Correction factor by which to adjust the weights
-
-    weights_corrected : array, unit = Jy^-2
-        Corrected weights assigned to observed visibilities, of the form
-        :math:`1 / \sigma^2`
+    weights : array,
+        Esimtate weight for each uv point.
+
+    Notes
+    -----
+        - This function does not support datasets where the weights may be
+          expected to be different for different baselines, e.g. if the data
+          includes visibilities from telescopes of different sizes
```

Comment:
What do you mean by `different weights for different baselines`?
By design, realistic observations are affected by baseline-dependent noise (weights).
Are you saying here that this function assumes that the scatter due to the noise is constant with baseline?
If so, I'd be more explicit.
If not, could you explain?

Reply:
The main thing is that when estimating the weights, any original weight information is thrown away.  This is probably fine for a utility function though.

Reply:
I'll edit the text.

Reply:
Hi Richard, I cannot see the change in the text.
You mean this docstring? Maybe you didn't comit push?

Reply:
Yeah, I forgot to push...